cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -170,11 +144,6 @@ public synchronized void startStopProgram(ProgramId programId, Map<String, Strin
    * @throws NotFoundException if the application to which this program belongs was not found
    */
   public ProgramStatus getProgramStatus(ProgramId programId) throws BadRequestException, NotFoundException {
-    // invalid type does not exist
-    if (programId.getType() == null) {
-      throw new BadRequestException(String.format("Invalid program type provided for program %s.", programId));
```

Comment:
So this method no longer throws `BadRequestException`, we should update javadoc and method signature.